### PR TITLE
Move tests to separate Android module and fix test failures

### PR DESCRIPTION
**Pull request by drewhannay**

### CODE REVIEW

File: dexmaker-tests/src/androidTest/java/com/android/dx/stock/ProxyBuilderTest.java [hunk 1]
```diff
@@ -685,18 +699,18 @@ public void testClassNotCachedWithDifferentParentClassLoaders() throws Exception
...
+
     public void testAbstractClassWithUndeclaredInterfaceMethod() throws Throwable {
         DeclaresInterface declaresInterface = proxyFor(DeclaresInterface.class)
                 .build();
         assertEquals("fake result", declaresInterface.call());
         try {
             ProxyBuilder.callSuper(declaresInterface, Callable.class.getMethod("call"));
             fail();
-        } catch (AbstractMethodError expected) {
+        } catch (IncompatibleClassChangeError expected) {
```
Comment (kkoser) [inline dexmaker-tests/src/androidTest/java/com/android/dx/stock/ProxyBuilderTest.java]: should we be catching both here? Did you try this on an old emulator?

File: dexmaker-tests/src/androidTest/java/com/android/dx/stock/ProxyBuilderTest.java [hunk 2]
```diff
@@ -984,7 +998,7 @@ private void doTestOrderClassWithDexMakerCaching() throws Exception {
         assertFalse(proxy.returnsBoolean());
         assertEquals(1.0, proxy.returnsDouble());
         assertNotNull(proxy.returnsObject());
-        assertEquals(2, versionedDxDir.listFiles().length);
+        assertTrue(versionedDxDir.listFiles().length != 0);
```
Comment (kkoser) [inline dexmaker-tests/src/androidTest/java/com/android/dx/stock/ProxyBuilderTest.java]: Should we maybe add a second check here, to make sure there are jar files? Not sure if its worth it.

File: dexmaker/src/main/java/com/android/dx/Code.java [hunk 3]
```diff
@@ -853,7 +855,7 @@ public void monitorEnter(Local<?> monitor) {
      * Releases the held lock on {@code monitor}.
      */
     public void monitorExit(Local<?> monitor) {
-        addInstruction(new ThrowingInsn(Rops.MONITOR_ENTER, sourcePosition,
+        addInstruction(new ThrowingInsn(Rops.MONITOR_EXIT, sourcePosition,
```
Comment (kkoser) [inline dexmaker/src/main/java/com/android/dx/Code.java]: Nice catch

File: dexmaker-tests/src/androidTest/java/com/android/dx/stock/ProxyBuilderTest.java (hunk 2 above)
Comment (drewhannay) [inline dexmaker-tests/src/androidTest/java/com/android/dx/stock/ProxyBuilderTest.java]: We should improve the tests in general, but i think this is sufficient for now

File: dexmaker-tests/src/androidTest/java/com/android/dx/stock/ProxyBuilderTest.java (hunk 1 above)
Comment (drewhannay) [inline dexmaker-tests/src/androidTest/java/com/android/dx/stock/ProxyBuilderTest.java]: `AbstractMethodError` extends `IncompatibleClassChangeError`

Dalvik throws `AbstractMethodError` and ART throws `IncompatibleClassChangeError`, so catching `IncompatibleClassChangeError` works for both

File: dexmaker/src/main/java/com/android/dx/Code.java (hunk 3 above)
Comment (drewhannay) [inline dexmaker/src/main/java/com/android/dx/Code.java]: 😞

Comment (swankjesse) [inline dexmaker/src/main/java/com/android/dx/Code.java]: Yikes!

**Pull request merged**